portality/models/account.py

```diff
@@ -152,12 +149,19 @@ def prep(self):
     @property
     def api_key(self):
         if self.has_role('api'):
-            return self.data.get('api_key', None)
+            # Return the stored api key if it's there, generate one if not (there should be one present if the role is)
```

Comment:
My thought at the time was that returning ```None``` usually means you don't have API access, and the missing key is an accident so should be fixed by generating a new one. You're right about debugging though, it'll be a bit of a moving target as it generates new ones until save.

So if someone expects to have API access but their key can't be validated, do we expect them to login and generate themselves a new one? I guess that's straightforward. Anyway, there shouldn't be many opportunity for an account with the api role to be decoupled from its api_key record in data.

as a side note, why can't I put a function call in the dict.get() method? when I tried to write ```self.data.get('api_key', self.generate_api_key()) it always ran the generate, whereas if the default arg was ```None``` it would return the key from the dict.